.github/workflows/test.yml

```diff
@@ -460,6 +510,7 @@ jobs:
     if: needs.determine_jobs.outputs.rust == 'true'
     strategy:
       fail-fast: false
+      runs-on: ${{ matrix.os }}
```

Comment:
There is a `runs-on: ${{ matrix.os.runner }}` a few lines below.

Also there is no `runs-on` in `strategy`